File: readability.py
count_letter = 0
count_word = 1
count_sentence = 0

text = input("Text: ")
text_length = len(text)

for i in range(text_length):
    if(text[i].isalpha()):
        count_letter+=1

    if (text[i].isspace()):
        count_word +=1

    if(text[i] == '.' or text[i] == '?' or text[i] == '!'):
        count_sentence +=1

calculation = (0.0588 * count_letter / count_word * 100) - (0.296 * count_sentence / count_word * 100) - 15.8 #to calculate Index

index = round(calculation)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
int main(void)
{
    string text = get_string("Text: ");

    int i = strlen(text);
    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int x = 0; x < i; x++)
    {
        char c = text[x];
        if (isalpha(c) != 0)
        {
            letters++;
        }

        if (c == ' ')

            words++;


        if (c == '.' || c == '!' || c == '?')

           sentences++;

    }

    words ++;

    float L = ((float)letters / (float)words) * 100;
    float s = ((float)sentences / (float)words) * 100;
    float subindex = 0.0588 * L - 0.296 * s - 15.8;
    int index = round(subindex);
    if (index > 16)

        printf("Grade 16+\n");

    else if (index < 1)

        printf("Before Grade 1\n");

    else

        printf("Grade %i\n", index);


}

